commands/fixgas.go

```diff
@@ -0,0 +1,107 @@
+package commands
+
+import (
+	"context"
+	"math"
+	"math/big"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/build"
+	"github.com/filecoin-project/lotus/chain/store"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/go-pg/pg/v10"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/sentinel-visor/lens"
+	messagemodel "github.com/filecoin-project/sentinel-visor/model/messages"
+	"github.com/filecoin-project/sentinel-visor/storage"
+)
+
+var FixGas = &cli.Command{
+	Name: "fixgas",
+	Action: func(cctx *cli.Context) error {
+		ctx, rctx, err := setupStorageAndAPI(cctx)
+		if err != nil {
+			return xerrors.Errorf("setup storage and api: %w", err)
+		}
+		defer func() {
+			rctx.closer()
+			if err := rctx.db.Close(ctx); err != nil {
+				log.Errorw("close database", "error", err)
+			}
+		}()
+
+		// Set up a context that is canceled when the command is interrupted
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		// Set up a signal handler to cancel the context
+		go func() {
+			interrupt := make(chan os.Signal, 1)
+			signal.Notify(interrupt, syscall.SIGTERM, syscall.SIGINT)
+			select {
+			case <-interrupt:
+				cancel()
+			case <-ctx.Done():
+			}
+		}()
+		return FixMessageGasEconomy(ctx, rctx.db, rctx.opener)
+	},
+}
+
+func FixMessageGasEconomy(ctx context.Context, db *storage.Database, opener lens.APIOpener) error {
+	api, closer, err := opener.Open(ctx)
+	if err != nil {
+		return err
+	}
+	defer closer()
+
+	var msgs []*messagemodel.MessageGasEconomy
+	if err := db.DB.RunInTransaction(ctx, func(tx *pg.Tx) error {
+		err := tx.ModelContext(ctx, &msgs).Select()
+		if err != nil {
+			return err
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	for _, msg := range msgs {
```

Comment:
There are 383k rows in this table. Maybe the select will work but I don't know about the update. Is there pagination support in go-pg?

Reply:
I'm not sure, looking now.